src/main/java/org/apache/sling/bundleresource/impl/BundleResource.java

```diff
@@ -138,17 +135,19 @@ public BundleResource(final ResourceResolver resourceResolver,
                 try {
                     final URL url = this.cache.getEntry(propsPath);
                     if (url != null) {
-                        final JsonObject obj = Json.createReader(url.openStream()).readObject();
-                        for(final Map.Entry<String, JsonValue> entry : obj.entrySet()) {
-                            final Object value = getValue(entry.getValue(), true);
-                            if ( value != null ) {
-                                if ( value instanceof Map ) {
-                                    if ( children == null ) {
-                                        children = new HashMap<>();
+                        try (JsonReader reader = Json.createReader(url.openStream())) {
```

Comment:
If this is a general refactoring, please raise a dedicated PR for it. Mixing multiple aspects in a single PR makes it harder to review.

Reply:
reverted